lib/elixir/lib/macro/env.ex

```diff
@@ -349,6 +349,10 @@ defmodule Macro.Env do
 
     * #{trace_option}
 
+    * `:resolver` - a two argument function to use instead of the default resolver. The function will be invoked
+    with first argument being `:functions` or `:macros` and the second being the module. It has to return
+    a list of function arity key value pairs
```

Comment:
Why does it receive the module as argument if we already know the module? Maybe we should call it `info_callback` and default to `fun Module:'__info__'/1` in Erlang (`&module.__info__/1` in Elixir)?

Reply:
I can't see it. Since you have touched the code more recently than I did, do you have examples?

Reply:
The default cannot be as simple as `fun Module:'__info__'/1`. The `functions` version falls back to `module_info(exports)` and filter generated `(module|behaviour)_info` functions

Reply:
I think that's fine. We can document that the info callback fails, it will fallback to using the Module, and you only need to guarantee it does not fail in your case. :)